src/presetsController.ts

```diff
@@ -0,0 +1,844 @@
+      }
+    );
+
+    log.debug(localize('start.selection.of.config.presets', 'Start selection of configure presets. Found {0} presets.', presets.length));
+
+    log.debug(localize('opening.config.preset.selection', 'Opening configure preset selection QuickPick'));
+    const placeHolder = localize('select.active.config.preset.placeholder', 'Select a configure preset for {0}', this.folder.name);
+    const chosenPreset = await this.showPresetSelector(presets, { placeHolder });
+    if (!chosenPreset) {
+      log.debug(localize('user.cancelled.config.preset.selection', 'User cancelled configure preset selection'));
+      return false;
+    } else if (chosenPreset === this._cmakeTools.configurePreset?.name) {
+      return true;
+    } else if (chosenPreset === '__addPreset__') {
+      await this.addConfigurePreset();
+      return false;
+    } else {
+      log.debug(localize('user.selected.config.preset', 'User selected configure preset {0}', JSON.stringify(chosenPreset)));
+      await this.setConfigurePreset(chosenPreset);
+      return true;
+    }
+  }
+
+  async setConfigurePreset(presetName: string): Promise<void> {
+    // Load the configure preset into the backend
+    await vscode.window.withProgress(
+        {
+          location: vscode.ProgressLocation.Notification,
+          title: localize('loading.config.preset', 'Loading configure preset {0}', presetName),
+        },
+        () => this._cmakeTools.setConfigurePreset(presetName),
+    );
+    await vscode.window.withProgress(
+        {
+          location: vscode.ProgressLocation.Notification,
+          title: localize('reloading.build.test.preset', 'Reloading build and test presets'),
+        },
+        async () => {
+          const buildPreset = this._cmakeTools.buildPreset?.name;
+          const testPreset = this._cmakeTools.testPreset?.name;
+          if (buildPreset) {
+            await this.setBuildPreset(buildPreset);
+          }
+          if (testPreset) {
+            await this.setTestPreset(testPreset);
+          }
+        }
+    );
+  }
+
+  private async checkConfigurePreset(): Promise<preset.ConfigurePreset | null> {
+    const selectedConfigurePreset = this._cmakeTools.configurePreset;
+    if (!selectedConfigurePreset) {
+      const message_noConfigurePreset = localize('config.preset.required', 'A configure preset needs to be selected. How would you like to proceed?');
+      const option_selectConfigurePreset = localize('select.config.preset', 'Select configure preset');
+      const option_later = localize('later', 'later');
+      const result = await vscode.window.showErrorMessage(message_noConfigurePreset, option_selectConfigurePreset, option_later);
+      if (result === option_selectConfigurePreset && await vscode.commands.executeCommand('cmake.selectConfigurePreset')) {
+        return this._cmakeTools.configurePreset;
+      }
+    }
+    return selectedConfigurePreset;
+  }
+
+  async selectBuildPreset(): Promise<boolean> {
```

Comment:
I ran into an issue where both my presets did not appear in the quick pick:
![image](https://user-images.githubusercontent.com/12818240/114224960-9dc9ff00-9926-11eb-84d0-51b5524b3492.png)

Reply:
Only the build presets with configurePreset set to the currently active configure preset will show up in the selector

Reply:
Haha, ok.  I get it.